sync.go

```diff
@@ -161,54 +159,44 @@ func (c *Sync) WriteMessage(message *Message, content *[]byte) error {
 
 // ReadMessage is a blocking function that will wait until a frisbee message is available and then return it (and its content).
 // In the event that the connection is closed, ReadMessage will return an error.
-func (c *Sync) ReadMessage() (*Message, *[]byte, error) {
+func (c *Sync) ReadMessage() (*packet.Packet, error) {
 	if c.closed.Load() {
-		return nil, nil, ConnectionClosed
+		return nil, ConnectionClosed
 	}
 	var encodedMessage [protocol.MessageSize]byte
 
 	_, err := io.ReadAtLeast(c.conn, encodedMessage[:], protocol.MessageSize)
 	if err != nil {
 		if c.closed.Load() {
-			c.logger.Error().Msgf(errors.WithContext(ConnectionClosed, POP).Error())
-			return nil, nil, errors.WithContext(ConnectionClosed, POP)
+			c.Logger().Error().Err(ConnectionClosed).Msg("error while reading from underlying net.Conn")
+			return nil, ConnectionClosed
 		}
-		c.logger.Error().Msgf(errors.WithContext(err, POP).Error())
-		return nil, nil, errors.WithContext(c.closeWithError(err), POP)
+		c.Logger().Error().Err(err).Msg("error while reading from underlying net.Conn")
+		return nil, c.closeWithError(err)
 	}
+	p := packet.Get()
```

Comment:
This is one of the clearest use cases for how we're using packet instaed of the `frisbee.Message` - now we just use `packet.Get()` to get a packet struct, decode the byte slices directly into it, read content from the `net.Conn` directly into the `packet.Content` byte slice, and when we use `packet.Put(p)` later, we are enabling the reuse of the `packet.Packet` struct *as well as* the `packet.Content` byte slice - it's a trick I stole from fasthttp and it SIGNIFICANTLY improves performance.